docs/diagrams/AddSequenceDiagram.puml

```diff
@@ -0,0 +1,117 @@
+@startuml
+
+box logic
+participant ":Parser" as Parser
+participant ":AddCommand" as AddCommand
+end box
+
+box template
+participant ":RcTemplate" as RcTemplate
+end box
+
+box component
+participant ":Capacitor" as Capacitor
+participant ":Component" as Component
+end box
+
+[-> Parser : parse("add parallel c 20")
+activate Parser
+
+Parser -> Parser : (prepareAdd("add parallel c 20"))
+activate Parser
+
+Parser -> AddCommand : AddCommand(template,"parallel","c", 500)
+activate AddCommand
+
+Parser <- AddCommand : add
+deactivate AddCommand
+
+Parser <- Parser : add
+deactivate Parser
+
+[<- Parser : add
+deactivate Parser
+
+[-> AddCommand: Execute
+activate AddCommand
+
+AddCommand -> RcTemplate : rc.getComponent(c)
+activate RcTemplate
+
+RcTemplate -> Capacitor : getComponent(c)
+activate Capacitor
+
+RcTemplate <- Capacitor : capacitor
+deactivate Capacitor
+
+AddCommand <- RcTemplate : capacitor
+deactivate RcTemplate
+
+AddCommand -> AddCommand : getNewValue()
```

Comment:
I realise that we might need to change the actual Java code here so it doesn't need to call `getComponent()` twice.

Reply:
Yeah it looked redundant to me as well, will make the edits after we append the code.